### ISSUE TITLE
Task- Tif files are too large to upload to Comprehensive specimen digitization

### ISSUE THREAD

**Issue by hah1009** [SpeciesFileGroup/taxonworks#2052]
Hello,
I was hoping to upload .tif files to collection objects through Comprehensive specimen digitization. When I did this I received an error message that the file (1.6 Gb) was too large. 

![Github 1](https://user-images.githubusercontent.com/63208032/108380454-d83fd700-71d4-11eb-982e-ea6765f71595.jpg)

I was wondering if future updates will allow for uploads of larger files, such as tif files, or if I should seek out a different way to upload high resolution images.

Thank you!

**Comment by mjy** [comment]
1.6 GB is a *huge* image. It's not impossible, but I'm curious, what was the source?

**Comment by hah1009** [comment]
I am working with tif files made with photos taken by a Canon 8.0V camera through a compound microscope. Each tif file is composed of  ~15-25 JPG images (each are a 3-5 Mb).

**Comment by mjy** [comment]
25 * 5 mb = 75 mb, not 1.6 GB?!

**Comment by hah1009** [comment]
Unfortunately I am not sure where this extra size is coming from. I am using FIJI to create the tif files, and I will look into what is causing this and how I can make them smaller.

**Comment by LocoDelAssembly** [comment]
> 25 * 5 mb = 75 mb, not 1.6 GB?!

But if then saved with lossless compression (or no compression at all) it might be possible (like a ~572MP image or so?).

**Comment by mjy** [comment]
If the 5mb images are lossless then 25 lossless images shouldn't add up even to 572MP, right?

Are you changing (perhaps inadvertently) the color space to something like 16bit when the source is 8?  

In theory any size is possible.  We can extend the limits. The main issue will be your upload speed, uploading that single image will take a long time.

**Comment by LocoDelAssembly** [comment]
The source images likely are lossy JPEGs, so all of them may fit perfectly in under 100MB even if 20MP each. However after composing all of them in a single TIFF with LZW compression (only compression guaranteed to be compatible with all readers) or no compression at all, then it will get may times bigger than original files combined.

**Comment by mjy** [comment]
Thanks @LocoDelAssembly. Does that assume image flattening in the Tiff?

**Comment by LocoDelAssembly** [comment]
Not sure if you can use layers but still use LZW or no compression, but I guess this TIFF is likely flattened with all JPEGs stitched possibly automatically.

**Comment by mjy** [comment]
@hah1009 is this a multi-focus or gig-pan style image?

**Comment by hah1009** [comment]
Hello,
Thank you all for your help.

This is not a multi-focus image, and I do not believe it is gig-pan style.

I tried using a different program to convert the images into a multi tiff file where I can select for LZW compression. This has greatly reduced the size of the files (thank you all for your input!). The files I have now are within the range of 182-239 Mb.

When I first uploaded the 1.6 GB file, the error message I received said that the max file size is 256 Mib. 

However, now when I try to upload a file size of 239 Mb, I receive another error message that the file is still too big. I also tried uploading a file that was 182 Mb and received the same result (see screenshot below). 

Please let me know if you would prefer that I create a separate issue for this, or if it is another problem on my end.

![Github 3](https://user-images.githubusercontent.com/63208032/108866006-ca69c780-75c1-11eb-8563-1f062f0c5467.jpg)

**Comment by mjy** [comment]
@LocoDelAssembly @dimus nginx setting it seems.  Can we bump to 256mb?

**Comment by LocoDelAssembly** [comment]
Lets try something else before resorting to yet more upload size. @hah1009 did the program offer the possibility to compress in JPEG? (Not compliant with all readers, but I think TaxonWorks can handle it). Like https://feedback.photoshop.com/conversations/photoshop/photoshop-cant-save-tiff-with-jpeg-compression/5f5f45a34b561a3d42574568?commentId=5f5f481c4b561a3d420a2eb7 but with the program you use.

**Comment by mjy** [comment]
Is that lossy?

**Comment by LocoDelAssembly** [comment]
It is, but if saved at high quality the damage should be close to non-existent, given that the source images used to compose the TIFF were JPEGs already. Saving them losslessly would only serve to encode the JPEG compression artifacts explicitly in the file, without really improving anything over saving with JPEG.

**Comment by hah1009** [comment]
Yes, I do have to option to compress in JPEG. I tried this with the file that was 182 Mb from the LZW compression. When compressed in JPEG it came out to be 1.13 Gb (the same size that FIJI produced).

**Comment by debpaul** [comment]
@jlpereira do we know if this is solved? for now?

**Comment by mjy** [comment]
@hah1009 can you let us know if it's still an issue for you?

**Comment by mjy** [comment]
Closing as out of date.
